torch/csrc/jit/tensorexpr/function.cpp

```diff
@@ -99,44 +127,110 @@ Tensor* Compute(
   }
   std::vector<const Expr*> dims;
   std::vector<const Var*> args_nodes;
-  unpack_dim_args(dim_args, &dims, &args_nodes);
+  DimArg::unpack_dim_args(dim_args, &dims, &args_nodes);
   auto args = VarVectorToVarHandleVector(args_nodes);
   const Expr* body = body_func(args[0], args[1], args[2], args[3]).node();
   Function* func = new Function(func_name, dims, args_nodes, body);
   return new Tensor(func, 0);
 }
 
-Stmt* Function::ElementStmt(size_t index) {
-  std::vector<ExprHandle> strides(dims_.size());
-  for (size_t i = 0; i < strides.size(); i++) {
-    if (i == strides.size() - 1) {
-      strides[i] = ExprHandle(1);
-      continue;
-    }
-    ExprHandle stride = ExprHandle(dims_[i + 1]);
-    for (size_t j = i + 2; j < dims_.size(); j++) {
-      stride = stride * ExprHandle(dims_[j]);
-    }
-    strides[i] = stride;
+Tensor* ReduceCommon(
+    const std::string& func_name,
+    const Dtype& dtype,
+    std::vector<const Expr*>& outer_dims,
+    std::vector<const Var*>& outer_args,
+    ExprHandle initializer,
```

Comment:
can we instead store identity values for supported ops?  we'll need to anyway for optimizations and can get rid of the initializer argument entirely.

sorry if this seems gratuitously pedantic, but I'm quite worried about code maintenance and potential misuse (e.g. initialized to a non-identity)

Reply:
The part of this I'm worried about is "supported ops". Like it fine to code in that Sum (the default reduction over Add) is based at zero, but is there a sensible way to do that for a compound expression? Even the simplest matmul case is a lambda of multiple ops.